common/persistence/cassandraPersistence.go

```diff
@@ -432,6 +472,16 @@ const (
 		`and visibility_ts = ? ` +
 		`and task_id = ? `
 
+	templateDeleteWorkflowExecutionSignalRequestedQuery = `UPDATE executions ` +
+		`SET signal_requested = signal_requested - ? ` +
+		`WHERE shard_id = ? ` +
+		`and type = ? ` +
+		`and domain_id = ? ` +
+		`and workflow_id = ? ` +
+		`and run_id = ? ` +
+		`and visibility_ts = ? ` +
+		`and task_id = ? `
```

Comment:
This is also updating the mutable state.  Make sure it is protected by conditional check on range_id and next_event_id.

Reply:
changed, protected by using updateWorkflowExecution.